### PR TITLE
Docs: Split client API into sub-pages

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.
Ready for review and merge.

Review is best done by looking at the built documentation, and particularly by looking at the begins of:
* each H1 heading, and
* each H2 heading of chapter 4.

### CODE REVIEW

**Comment by KSchopmeyer** [comment]
Note that I made some general comments on wording that are outside of these changes.  Feel free just to start another issue rather than worrying about them here.

I do see, however, that putting detailed info into the rst leads to it being forgotten.  I had even forgottien that with the logging config changes.

**Comment by andy-maier** [comment]
I'd like to resolve your comments about the content in another PR, but this PR still has some things related to the chapter-level TOC and the left-hand menu items for chapter 4.

**Comment by KSchopmeyer** [comment]
Sorry for the really poor formatting of the big group of comments above.   Here they are repeated but formatted so that they are readable.  Apparently you cannot edit them once you put them in:

The following are general comments from working through the built documentation and don't apply directly to one of the edits.

**1.** CIM_Operations.py: note after operations table the word servers should be WBEM servers

**2.**   Section 7, MOF Compiler: The first sentence (I assume from mof_compiler.py) is really vapid

"A MOF compiler."

Should probably almost a duplicate of the second sentence of the compiler.rst section. That tells them what the compiler is and what it does. In fact, I wonder if we should move most of that lead in to the mof_compiler.py file since it has minimal formatting but is a good overview of the compiler and its limitations.

GENERAL COMMMENT: I personally find that things we put into just the doc we forget about. Whereas if it is in the doc string of the python files, we look at it occasionally. I am leaning more and more to put the good stuff in the doc string unless it is a formatting issue.  Then we will tend to review it and keep it up to date.

**3.**  Couple of general suggestions on section titles:

**3.1.** Change the section title on statistics section to WBEM Operation Statistics

**3.2.** Change the section title on Logging section to WBEM Operation Logging

**3.3.** Change title on Value mappings to Value mapping of CIM ValueMap/Value Qualifiers.

Since we no longer have the little lead description of what is in Section 4in to WBEM Client as the first thing shown in the web page when you click on Section 4 the titles probably need to be more expressive. Now the user has to go just by the name without the extra explanation sentence.

This is the old hand build menu that used to show up. as the first thing on the section 4 web page.

``
    WBEM operations - Class WBEMConnection is the main class of the WBEM client library API and is used to issue WBEM operations to a WBEM server.
    WBEM Operation recording - Class WBEMConnection includes the capability to optionally record the 
        WBEMConnection method calls that communicate with a WBEM server including creating yaml files for test generation and logging using the Python logging facility.
    CIM objects - Python classes for representing CIM objects (instances, classes, properties, etc.) that are used by the WBEM operations as input or output.
    CIM data types - Python classes for representing values of CIM data types.
Conversion functions - Conversion functions related to CIM objects and CIM data types.
    CIM status codes - CIM status codes returned by failing WBEM operations.
    Logging - Logging of WBEM operations.
    Exceptions - Exceptions specific to pywbem that may be raised.
    Statistics - Statistics classes to gather information on wbem operation performance.
    WBEM Operation recorder - Python classes that implement the operation recorder functions that are used by WBEMConnection.
    Value mappings - Utility class for mapping the values of CIM elements qualified with ValueMap and Values.
    Security considerations - Information about authentication types and certificates.

**4.**  The left-side panel - What appears in that panel is different for section 4 than for other sections.
When I click on a major section (ex. section 4. I was comparing the web pages between latest on RTD and this and in RTD when I click on a major section it shows the whole structure of that section (ex. 4, 4.1, 4.1.1, 4.1.2, 4.2, etc.) Not it only shows the Previous Next, etc. and the subsections on the web page itself (4.1, 4.2. 4.3, etc).

In conclusion, the left side menu does different things depending on which section is selected. All of the sections except section 4 appear to show the "Table of Contents" with the current section in the left-panel. Section 4 shows the Previous / Next topic menu in the left pane.

This is apparently an issue with section 4 since section 1 does show the full set of subsections of section 1 in the left side menu.

Part of the problem is that when section 4 is selected and the left-side menu only shows next, prev, etc. It does not show the Table of Contents button so the user has to sort of back out of that page with the back button

I wonder if part of this is the :depth 2 attribute that you added to the rst attributes.

**5.** When the left-side panel is on Section 4 or section 4.1 it shows 4.1.1 but not 4.1.2

**6.** First sentence in cim_obj.py module doc:
    "Python classes for representing values of CIM data types, and related conversion functions."

Should we note that there are also special conversion functions defined in the section on conversion functions.

So far I have gone through most of the sections comparing for issues in the lead in paragraphs, and checking that in those sections where there are detailed items for each class, the same sections show up in both the  new built document and the old document. No problems that I could see.

I have not spent much time on logging, the recorder, or the mocker yet and I assume that Development section has not changed.

The big issue as stated above is that the the left-side panel seems to be much less useful now, in particular for section 4 since it shows the Next/Previous menu and not the section tree menu.

File: docs/client/conversion.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+.. _`Conversion functions`:
+
+Conversion functions
+--------------------
+
+This section describes conversion functions related to :ref:`CIM objects` and
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/client/conversion.rst]: This comment is now pursued in issue #1111.

File: docs/client/recording.rst [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+This is disabled by default and can be enabled by adding recorders to a
+WBEMConnection with the method
+:meth:`~pywbem.WBEMConnection.add_operation_recorder`.
+
+Typical usage scenarios for various operation recorders are the tracing of WBEM
+operations, tracing of interactions with a WBEM server, or the generation of
+test cases.
+
+Please not that the method of activating operation recorders changed starting
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/client/recording.rst]: This comment is now pursued in issue #1111.

**Comment by andy-maier** [comment]
Karl, sometimes the pen icon for editing a comment is not shown, but then there is an "Edit" item in the menu that comes up when you click on ". . . ".

**Comment by andy-maier** [comment]
Because this PR is the basis for some other PRs, I decided to resolve the comments that were made via the newly created issue #1111, and to merge this PR to make progress.

**Pull request merged**